### ISSUE TITLE
Add kubernetes-worker config option for enabling privileged containers

### ISSUE THREAD

**Issue by wwwtyro** [juju-solutions/bundle-canonical-kubernetes#223]
Would enable/disable/automate `--allow-privileged` option for kubelet.

Needed to prevent e2e test failure, at a minimum.

**Comment by juledwar** [comment]
Same requirement - I'm blocked on this as I need privileged containers.

**Comment by wwwtyro** [comment]
@juledwar @rimusz Just want to let you guys know that we are working on this.

@tvansteenburgh Do you think you'll do this as part of your work? If not or if you don't have time, hit me up.

**Comment by tvansteenburgh** [comment]
@wwwtyro I can roll it into my gpu branch.

**Comment by tvansteenburgh** [comment]
New option will look like this:

```yaml

  allow-privileged:
    type: string
    default: "false"
    description: |
      Allow privileged containers to run on worker nodes. Supported values are
      "true", "false", and "auto". If "true", kubelet will run in privileged
      mode by default. If "false", kubelet will never run in privileged mode.
      If "auto", kubelet will not run in privileged mode by default, but will
      switch to privileged mode if gpu hardware is detected.
```

I guess we should do this for kubernetes-master too while we're at it? kube-apiserver needs to run in privileged mode for gpu support. cc @marcoceppi

**Comment by Cynerva** [comment]
Should "auto" be the default instead? I think one of our goals was to have GPU support work without having to override the default config.

**Comment by tvansteenburgh** [comment]
@Cynerva Good point, I'll switch that.

**Comment by tvansteenburgh** [comment]
This is baked into https://github.com/kubernetes/kubernetes/pull/43467

**Comment by Southclaws** [comment]
Running into this issue, I've tried to use `cs:bundle/canonical-kubernetes-21` and I'm getting a `Forbidden: disallowed by policy` because of no `--allow-privileged` flag. I've also tried the bundle linked in kubernetes/kubernetes#43467 `cs:~tvansteenburgh/bundle/kubernetes-gpu-support-3` which resolves the privileged issue but I'm trying to deploy [Pachyderm](https://github.com/pachyderm/pachyderm) which does not support Kubernetes 1.6 yet.

**Comment by chuckbutler** [comment]
@Southclaws @juledwar  this was released with our 1.6.1 release of the CDK/Kubernetes-Core bundles. I'm  going to close for now, if you encounter any further issues please reply and we'll re-open and evlauate for a fix.
